x/compliancetest/types/message_add_testing_result_test.go

```diff
@@ -1,42 +1,135 @@
 	}{
 		{
 			name: "invalid address",
 			msg: MsgAddTestingResult{
-				Signer: "invalid_address",
+				Signer:                "invalid_address",
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
 			},
 			err: sdkerrors.ErrInvalidAddress,
-		}, {
+		},
+		{
+			name: "vid is 0",
+			msg: MsgAddTestingResult{
+				Signer:                sample.AccAddress(),
+				Vid:                   0,
+				Pid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "vid < 0",
+			msg: MsgAddTestingResult{
+				Signer:                sample.AccAddress(),
+				Vid:                   -1,
+				Pid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "vid > 65535",
+			msg: MsgAddTestingResult{
+				Signer:                sample.AccAddress(),
+				Vid:                   65536,
+				Pid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
+			},
+			err: validator.ErrFieldUpperBoundViolated,
+		},
+		{
+			name: "pid is 0",
+			msg: MsgAddTestingResult{
+				Signer:                sample.AccAddress(),
+				Pid:                   0,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "pid < 0",
+			msg: MsgAddTestingResult{
+				Signer:                sample.AccAddress(),
+				Pid:                   -1,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "pid > 65535",
+			msg: MsgAddTestingResult{
+				Signer:                sample.AccAddress(),
+				Pid:                   65536,
+				Vid:                   1,
+				SoftwareVersionString: testconstants.SoftwareVersionString,
+				TestDate:              testconstants.CertificationDate.Format(time.RFC3339),
+				TestResult:            testconstants.TestResult,
+			},
+			err: validator.ErrFieldUpperBoundViolated,
```

Comment:
Please also add the following negative tests:
1) TestDate is not set (empty string). Expected: ErrRequiredFieldMissing
2) TestResult is not set (empty string). Expected: ErrRequiredFieldMissing
3) TestDate doesn't have expected format RFC3339)